src/main/java/com/kodilla/ecommercee/scheduler/EmailScheduler.java

```diff
@@ -29,7 +28,8 @@ public void sendEmail(){
         List<UserOrder> userOrdersToSendEmail = orderRepository.findByMailSentFalse();
         for(UserOrder userOrder: userOrdersToSendEmail){
             emailService.send(new Mail(
-                    adminConfig.getAdminMail(),SUBJECT,"New order for user "+ userOrder.getUser().getName()));
+                    adminConfig.getAdminMail(), SUBJECT, "New order for user " + userOrder.getUser().getName()));
+            orderRepository.setMailSent(userOrder.getId(), true);
```

Comment:
Poprzednio nie pisałem, ale może zamiast tej sporej kwerendy coś takiego:
userOrder.setMailSent(true);
orderRepository.save(userOrder);
To tylko luźna sugestia, z Twoim rozwiązaniem testy przechodzą (sprawdziłem), więc jest ok.
Brakuje tylko rozwiązania, co w sytuacji, jeśli wysłanie maila zakończy się błędem,

Reply:
ok słuszne uwagi, poprawie to w przyszłości ale chwilowo chce wrzucić zmiany bo jest ich dużo aby też z pracą innych się zgadzało